src/ui/features/turnstile/Turnstile.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import { useWindowSizeStore } from 'src/ui/shared/useWindowSizeStore';
+import { invariant } from 'src/shared/invariant';
+import { emitter } from 'src/ui/shared/events';
+import { isObj } from 'src/shared/isObj';
+import type { HTMLDialogElementInterface } from 'src/ui/ui-kit/ModalDialogs/HTMLDialogElementInterface';
+import { BottomSheetDialog } from 'src/ui/ui-kit/ModalDialogs/BottomSheetDialog';
+import { queryClient } from 'src/ui/shared/requests/queryClient';
+import { WebAppMessageHandler } from '../referral-program/WebAppMessageHandler';
+
+export function TurnstileTokenHandler() {
+  const { innerWidth } = useWindowSizeStore();
+  const turnstileWidgetHeight = 73;
+  const turnstileWidgetWidth = innerWidth - 32;
+  const [showDialog, setShowDialog] = useState(false);
+  const dialogRef = useRef<HTMLDialogElementInterface | null>(null);
+
+  useEffect(() => {
+    return emitter.on('openTurnstile', () => {
+      setShowDialog(true);
+      dialogRef.current?.showModal();
+    });
+  }, []);
```

Comment:
We shouldn't be calling any unnecessary hooks and rendering any unnecessary jsx unless we are reacting to this event. The component should be split into two, and the outer one should only be listening to this event and rendering `null` until the event occurs.

Reply:
Agree! I will update this!